src/main/java/net/dv8tion/jda/internal/utils/LRUCachePolicy.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015-2020 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.utils;
+
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.utils.MemberCachePolicy;
+
+import javax.annotation.Nonnull;
+import java.util.LinkedList;
+
+public class LRUCachePolicy implements MemberCachePolicy
+{
+    private final int max;
+    private final LinkedList<Long> cached = new LinkedList<>(); // TLongLinkedList had a bug so we gotta use this
+
+    public LRUCachePolicy(int max)
```

Comment:
Should we consider supporting this on a per-guild basis instead of all members globally?

Reply:
This is more of a concept implementation tbh. Users can implement something more advanced.

Reply:
I wonder if we should move this to the examples instead then? I don't have a strong opinion either direction, just a thought.